testing/e2e/agent_install_test.go

```diff
@@ -0,0 +1,348 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+//go:build e2e
+
+package e2e
+
+import (
+	"archive/tar"
+	"archive/zip"
+	"bytes"
+	"compress/gzip"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/elastic/fleet-server/v7/version"
+
+	"github.com/stretchr/testify/suite"
+)
+
+type AgentInstallSuite struct {
```

Comment:
I have not been able run this testsuite yet as docker  in my vm is giving me issues, can anyone help validate?

Reply:
Tried to execute, but there seems to be some version mismatch :thinking: 
```
$ make test-e2e
...
 > COPY --chown=elastic-agent cover/fleet-server-8.8.0-SNAPSHOT-linux-x86_64/fleet-server              ./data/elastic-agent-43a847/components/fleet-server:
------
Dockerfile:1
--------------------
   1 | >>> ARG ELASTIC_AGENT_IMAGE # e.g. docker.elastic.co/cloud-release/elastic-agent-cloud:8.8.0-4671daa2-SNAPSHOT
   2 |     
   3 |     FROM --platform=linux/amd64 ${ELASTIC_AGENT_IMAGE} as elastic_agent_amd64
--------------------
ERROR: failed to solve: failed to compute cache key: failed to calculate checksum of ref moby::we2ydhofljha764g0lcfg4l7e: failed to walk /var/lib/docker/tmp/buildkit-mount777288482/cover/fleet-server-8.8.0-SNAPSHOT-linux-x86_64: lstat /var/lib/docker/tmp/buildkit-mount777288482/cover/fleet-server-8.8.0-SNAPSHOT-linux-x86_64: no such file or directory
make: *** [Makefile:323: build-e2e-agent-image] Error 1
```
It seems it builds version 8.9.0, but expects 8.8.0.
```
$ ls build/cover/
fleet-server-8.9.0-SNAPSHOT-linux-x86_64
```

Reply:
I just resolved an issue with the automated stack updates to `integration/.env`; ut just updated to 8.9.0-SNASHOT, i'll merge main into this. Hopefully that resolves it

Reply:
I'll work on the agent_install_test at another time, there's a lot of edge cases that will make this pr much too large